pkg/config/config_template.yaml

```diff
@@ -1,858 +1,1197 @@
+#
+# statsd_forward_host: ""
+
+## @param statsd_forward_port - integer - optional - default: 0
+## Port or the "statsd_forward_host" to forward StatsD packet to.
+#
+# statsd_forward_port: 0
+
+## @param statsd_metric_namespace - string - optional - default: ""
+## Set a namespace for all StatsD metrics coming from this host.
+## Each metric received is prefixed with the namespace before it's sent to Datadog.
+#
+# statsd_metric_namespace: ""
+
 {{ end -}}
-{{- if .JMX }}
-# JMX
+{{- if .Metadata }}
+
+## @param metadata_providers - list of custom object - optional
+## Metadata providers, add or remove from the list to enable or disable collection.
+## Intervals are expressed in seconds. You can also set a provider's interval to 0
+## to disable it.
 #
-# jmx_pipe_path:
-# jmx_pipe_name: dd-auto_discovery
+# metadata_providers:
+#   - name: k8s
+#     interval: 60
+
+{{ end -}}
+{{- if .JMX }}
+
+#######################
+## JMX Configuration ##
+#######################
+
+## @param jmx_custom_jars - list of strings - optional
+## If you only run Autodiscovery tests, jmxfetch might fail to pick up custom_jar_paths
+## set in the check templates. If that is the case, force custom jars here.
 #
-# If you only run Autodiscovery tests, jmxfetch might fail to pick up custom_jar_paths
-# set in the check templates. If that is the case, you can force custom jars here.
 # jmx_custom_jars:
 #   - /jmx-jars/jboss-cli-client.jar
+
+## @param jmx_use_cgroup_memory_limit - boolean - optional - default: false
+## When running in a memory cgroup, openjdk 8u131 and higher can automatically adjust
+## its heap memory usage in accordance to the cgroup/container's memory limit.
+## The Agent set a Xmx of 200MB if none is configured.
+## Note: older openjdk versions and other jvms might fail to start if this option is set
 #
-# When running in a memory cgroup, openjdk 8u131 and higher can automatically adjust
-# its heap memory usage in accordance to the cgroup/container's memory limit.
-# Default is false: we'll set a Xmx of 200MB if none is configured.
-# Note: older openjdk versions and other jvms might fail to start if this option is set
-#
-# jmx_use_cgroup_memory_limit: true
+# jmx_use_cgroup_memory_limit: false
+
+## @param jmx_max_restarts - integer - optional - default: 3
+## Number of JMX restarts allowed in the restart-interval before giving up.
 #
-# Number of JMX restarts allowed in the restart-interval before giving up
 # jmx_max_restarts: 3
+
+## @param jmx_restart_interval - integer - optional - default: 5
+## Duration of the restart interval in seconds.
 #
-# Duration of the restart interval in seconds
 # jmx_restart_interval: 5
+
+## @param jmx_thread_pool_size - integer - optional - default: 3
+## JMXFetch collects multiples instances concurrently. The following options may
+## help fine-tune the level of concurrency and timeouts that come into play during the
+## collection of metrics from configured instances:
```

Comment:
this paragraph actually applies to a few of the following options, not only `jmx_thread_pool_size`. Not sure if there's a way to reflect that

Reply:
which ones?